### PR TITLE
Don't play with pipefail

### PR DESCRIPTION
**Pull request by mkolesnik**
Pipefail is already globally set on travis, no need to play with it
since turning it off at the end causes e2e failure to be missed.

Also makes sense to set it at the beginning of the script to make sure
the entire experience is consistent wvwn without the CI.

### CODE REVIEW

File: scripts/kind-e2e/e2e.sh [hunk 1]
```diff
@@ -187,6 +187,8 @@ with_context cluster2 connectivity_tests
 
 # dataplane E2E need to be modified for globalnet
 if [[ $globalnet = false ]]; then
+    unset KUBECONFIG
```
Comment (dfarrell07) [inline scripts/kind-e2e/e2e.sh:190]: This is related to pipefail? Or just some random cleanup? (which is fine, but wanted to make sure you meant to do it here:)

Comment (mkolesnik) [reply scripts/kind-e2e/e2e.sh:190]: Indeed it is, please look at the specific commit to understand why :) - https://github.com/submariner-io/submariner-operator/pull/317/commits/c1d27c71b460eb0248814e75aaa9b511bc159139

Comment (dfarrell07) [reply scripts/kind-e2e/e2e.sh:190]: Ah I see your explanation, thanks!

File: scripts/kind-e2e/e2e.sh [hunk 2]
```diff
@@ -184,6 +184,8 @@ with_context cluster2 connectivity_tests
...
```
Comment (skitt) [inline scripts/kind-e2e/e2e.sh:190]: This is fixed in https://github.com/submariner-io/submariner-operator/pull/314.

Comment (mkolesnik) [reply scripts/kind-e2e/e2e.sh:190]: Awesome, can we get that merged then?

File: scripts/kind-e2e/e2e.sh [hunk 3]
```diff
@@ -153,7 +150,7 @@ fi
 PRJ_ROOT=$(git rev-parse --show-toplevel)
 SUBMARINER_BROKER_NS=submariner-k8s-broker
 # FIXME: This can change and break re-running deployments
-SUBMARINER_PSK=$(cat /dev/urandom | LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1)
+SUBMARINER_PSK=$(head -c 10000 /dev/urandom | LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1)
```
[the remark this answers is not in the document]
Comment (mkolesnik) [reply scripts/kind-e2e/e2e.sh]: Thanks, I've committed your suggestion

File: scripts/kind-e2e/lib_operator_verify_subm.sh [hunk 4]
```diff
@@ -35,7 +35,11 @@ function verify_subm_operator() {
 function validate_equals() {
   local json_filter=$1
   local expected=$2
-  [[ $expected = $(jq -r -M $json_filter $json_file) ]]
+  local actual=$(jq -r -M $json_filter $json_file)
+  if [[ $expected != $actual ]]; then
+     echo "Expected ${expected@Q} but got ${actual@Q}"
```
[the remark this answers is not in the document]
Comment (mkolesnik) [reply scripts/kind-e2e/lib_operator_verify_subm.sh:79]: Sure :)
Had to do it during the investigation to identify the failures..

**Pull request merged**